horace_core/sqw/@sqw/unary_op_manager.m

```diff
@@ -0,0 +1,23 @@
+function w = unary_op_manager(w1, unary_op)
+% Implement unary arithmetic operations for objects containing a signal and variance arrays.
+
+% Generic method edited for sqw class. Must have
+%   (1) have methods to set, get and find size of signal and variance arrays:
+%           >> sz = sigvar_size(obj)
+%           >> w = sigvar(obj)          % w is sigvar object (has fields w.s, w.e)
+%           >> obj = sigvar_set(obj,w)  % w is sigvar object
+%   (2) have dimensions method that gives the dimensionality of the double array
+%           >> nd = dimensions(obj)
+%   (3) have private function that returns class name
+%           >> name = classname     % no argument - gets called by its association with the class
+
+w = w1;
+for i=1:numel(w1)
+    if ~isempty(w1(i).data.pix)
+        w(i).data.pix = w(i).data.pix.do_unary_op(unary_op);
+        w(i) = recompute_bin_data(w(i));
+    else
+        result = unary_op(sigvar(w1(i).data.s, w1(i).data.e));
+        w(i) = sigvar_set(w(i),result);
+    end
```

Comment:
would also be good to specify which sigvar this is - guessing that it is the Herbert utilities class taking 2 args. There are too many sigvars in the codebase